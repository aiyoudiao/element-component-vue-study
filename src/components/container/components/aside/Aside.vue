<script setup lang="ts">
import { withDefaults } from 'vue'
interface AsideType {
  width?: string
}

withDefaults(defineProps<AsideType>(), {
  width: '300px',
})
</script>

<script lang="ts">
export default {
  name: 'ElAside',
}
</script>

<template>
  <aside class="el-aside" :style="{ width }">
    <slot />
  </aside>
</template>

<style lang="scss">
@import '../../../../theme/src/aside.scss';
</style>
