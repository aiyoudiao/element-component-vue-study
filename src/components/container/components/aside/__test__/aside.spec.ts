import { mount } from '@vue/test-utils'
import Aside from '../Aside.vue'
describe('Aside.vue', () => {
  it('Aside snapshot', () => {
    const wrapper = mount(Aside)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('Aside render', () => {
    const wrapper = mount(Aside)
    expect(wrapper.classes()).toContain('el-aside')
  })
  it('Aside slot', () => {
    const wrapper = mount(Aside, {
      slots: {
        default: () => 'hello world',
      },
    })
    expect(wrapper.text()).toBe('hello world')
  })
  it('Aside style', () => {
    const wrapper = mount(Aside, {
      props: {
        width: '250px',
      },
    })
    expect(wrapper.attributes().style).toBe('width: 250px;')
  })
})

