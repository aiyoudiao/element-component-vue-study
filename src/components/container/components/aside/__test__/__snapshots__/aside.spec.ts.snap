// Vitest Snapshot v1

exports[`Aside.vue > Aside snapshot 1`] = `
<aside
  class="el-aside"
  style="width: 300px;"
>
  
  
</aside>
`;
